### PR TITLE
NMS-9771: Restructure and improve Help/Support pages

### PR DESCRIPTION
**Pull request by indigo423**
The support page contains the web form which is used by OpenNMS administrators to open commercial support tickets. The account. This page is now restricted to Admin users. Introduced a Help page which gives information where to find documentation and to get in contact with people. To give users a better visibility what version of OpenNMS is running, the version number is added to the footer page. To make it easier getting version information about the running Open "About" page in the user's menu is added. The web site components are refactored to be reusable as JSP includes.

JIRA: https://issues.opennms.org/browse/NMS-9771

### CODE REVIEW

File: opennms-webapp/src/main/webapp/about/index.jsp [hunk 1]
```diff
@@ -64,18 +64,17 @@
...
   <div class="panel panel-default">
     <div class="panel-heading">
-      <h3 class="panel-title">OpenNMS Web Console</h3>
+      <h3 class="panel-title">Version Details</h3>
     </div>
 <table class="table table-condensed">
   <tr>
     <th>Version:</th>
-    <td><%=Vault.getProperty("version.display")%></td>
+    <td><a href="http://docs.opennms.org/opennms/releases/latest/releasenotes/releasenotes.html" target="_blank" title="Show me the release Release Notes"><%=Vault.getProperty("version.display")%></a></td>
```
Comment (soleger) [inline opennms-webapp/src/main/webapp/about/index.jsp]: Please change title to "Release Notes".

File: opennms-webapp/src/main/webapp/includes/bootstrap-footer.jsp [hunk 2]
```diff
@@ -62,10 +63,10 @@
 
     <footer id="footer">
       <p>
-        OpenNMS <a href="support/about.jsp">Copyright</a> &copy; 2002-2017
+        OpenNMS <a href="about/index.jsp">Copyright</a> &copy; 2002-2017
         <a href="http://www.opennms.com/">The OpenNMS Group, Inc.</a>
         OpenNMS&reg; is a registered trademark of
-        <a href="http://www.opennms.com">The OpenNMS Group, Inc.</a>
+        <a href="http://www.opennms.com">The OpenNMS Group, Inc.</a> - Version: <%=Vault.getProperty("version.display")%>
```
Comment (soleger) [inline opennms-webapp/src/main/webapp/includes/bootstrap-footer.jsp]: Can you add a screenshot of what this looks like?

File: opennms-webapp/src/main/webapp/includes/bootstrap-footer.jsp [hunk 3]
```diff
@@ -43,8 +43,9 @@
 <%@page language="java"
 	contentType="text/html"
 	session="true"
-	import="java.io.File,org.opennms.web.api.HtmlInjectHandler"
+	import="java.io.File,org.opennms.core.resource.Vault"
 %>
+<%@ page import="org.opennms.web.api.HtmlInjectHandler" %>
```
Comment (soleger) [inline opennms-webapp/src/main/webapp/includes/bootstrap-footer.jsp]: Please use a single `import` statement.

File: opennms-webapp/src/main/webapp/includes/help-software-management.jsp [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+<div class="panel panel-default">
+    <div class="panel-heading">
+        <h3 class="panel-title">Software Management</h3>
+    </div>
+    <div class="panel-body">
+        <table class="table">
+            <tr>
+                <td style="border-top: none;"><a href="https://github.com/OpenNMS/opennms" target="_blank" class="btn btn-default" role="button" style="width: 100%">OpenNMS on GitHub</a></td>
+                <td style="border-top: none;">Our source code is version controlled with git and hosted on GitHub. Download the source code from latest develop or released versions of OpenNMS.</td>
```
Comment (soleger) [inline opennms-webapp/src/main/webapp/includes/help-software-management.jsp]: Change:
`Download the source code from latest develop or released versions of OpenNMS.`
to
`Download the source code for released versions of OpenNMS or the latest development code.`

File: opennms-webapp/src/main/webapp/includes/support-system-diagnostics.jsp [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+<div class="panel panel-default">
+    <div class="panel-heading">
+        <h3 class="panel-title">System Diagnostics</h3>
+    </div>
+    <div class="panel-body">
+        <table class="table">
+            <tr>
+                <td style="border-top: none;"><a href="admin/support/systemReport.htm" target="_blank" class="btn btn-default" role="button" style="width: 100%">Generate System Report</a></td>
+                <td style="border-top: none;">Generate support friendly information about your OpenNMS instance and system environment.</td>
```
Comment (soleger) [inline opennms-webapp/src/main/webapp/includes/support-system-diagnostics.jsp]: Hyphenate "support-friendly".

**Comment by mvrueden** [comment]
I think we should also link to the online-documentation of the release and not always to `latest`.

File: opennms-webapp/src/main/webapp/includes/bootstrap-footer.jsp (hunk 2 above)
Comment (indigo423) [reply opennms-webapp/src/main/webapp/includes/bootstrap-footer.jsp]: I have added detailed screenshots in the JIRA issue. I've tried to use as much plain bootstrap components as possible.

**Comment by indigo423** [comment]
@mvrueden can't fix this in this issue. @RangerRick needs to adjust docs deployment to have unified links with version numbers for releases, development and feature branches.

**Comment by soleger** [comment]
> unified links with version numbers for releases, development and feature branches

I would say that this only matters for releases and you should be able to use the version to create the link, right?

**Comment by indigo423** [comment]
@soleger we have problems with version numbers like "22.0.0-SNAPSHOT" and feature/branches. IMHO we can use the `version.display` but the links will then only for released versions.

File: opennms-webapp/src/main/webapp/includes/help-contact.jsp [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    </div>
+    <div class="panel-body">
+        <table class="table">
+            <tr>
+                <td style="border-top: none;"><a href="https://www.opennms.com/support/" target="_blank" class="btn btn-default" role="button" style="width: 100%">Commercial Support</a></td>
+                <td style="border-top: none;">Get commercial support for OpenNMS software provided by The OpenNMS Group, Inc.</td>
+            </tr>
+            <tr>
+                <td style="border-top: none;"><a href="https://chat.opennms.org/opennms" target="_blank" class="btn btn-default" role="button" style="width: 100%">Web Chat</a></td>
+                <td style="border-top: none;">Use the web based chat platform powered by Mattermost. The Discussion channel is a good starting point to reach users and developers. If you prefer IRC, you can join the "#opennms" channel on irc.freenode.net which is bridged to our Web Chat.</td>
```
Comment (soleger) [inline opennms-webapp/src/main/webapp/includes/help-contact.jsp]: Hyphenate "web-based".

File: opennms-webapp/src/main/webapp/includes/help-contact.jsp [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+            </tr>
+            <tr>
+                <td style="border-top: none;"><a href="https://wiki.opennms.org/wiki/Mailing_lists" target="_blank" class="btn btn-default" role="button" style="width: 100%">Mailing Lists</a></td>
+                <td style="border-top: none;">Get into contact with OpenNMS users in our community through our topic-driven mailing lists. Subsribing the opennms-discuss list is a good starting to ask question and how to get started with OpenNMS.</td>
```
Comment (soleger) [inline opennms-webapp/src/main/webapp/includes/help-contact.jsp]: "Subsribing the opennms-discuss list is a good starting to ask question and how to get started with OpenNMS." -> "The opennms-discuss mailing list is a good way to ask questions about getting started with OpenNMS."

**Comment by indigo423** [comment]
... any news here for review? @mvrueden @soleger branch and tests are green and review comments are in.

**Comment by j-white** [comment]
LGTM. Why do we need this in `foundation-2017` though?

**Comment by mvrueden** [comment]
The `version.display` would not work in all cases but it would work for all releases, so I'd vote to change it, even it is broken for some branches. We can also add an issue for this and address it another time.

**Comment by indigo423** [comment]
Rebased to develop.

**Comment by indigo423** [comment]
Added link to version specific online documentation. Anything else in this PR which needs to be changed?

**Comment by indigo423** [comment]
tests went green, anything else we need to do for this PR to apply any veto?

**Pull request merged**